File: PrimeiraAplicacao.Funcional/5_Tipos/Tuplas.fs
module Tuplas

let tupla1 = 1, 2, 3
let tupla2 = 1, true, "Renato"
let tupla3 = 3, false
let tupla4 = 1, 2, 3, false, "Teste"
let tupla5 = (+), false
let tupla6 = false, 2

let soma numero1 numero2 = 
    numero1 + numero2

let somaTupla (numero1, numero2) =
    numero1 + numero2

let item1, item2, item3 = tupla1

let item4, _, item5 = tupla2

let item6 = fst tupla3
let item7 = snd tupla3

let somarNumeroEInverterBool (numero, bool) = 
    numero + 5, not bool

let tupla3Modificada = somarNumeroEInverterBool tupla3

File: PrimeiraAplicacao.Funcional/Tipos/Tuplas.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Tipos
{
    class Tuplas
    {
        public void CriandoTuplas()
        {
            Tuple<int, int, int> tupla1 = new Tuple<int, int, int>(1, 2, 3);

            //Extraindo os itens da Tupla
            int item1 = tupla1.Item1;
            int item2 = tupla1.Item2;
            int item3 = tupla1.Item3;

            Tuple<int, bool, string> tupla2 =
                new Tuple<int, bool, string>(1, true, "Renato");

            Tuple<int, bool> tupla3 = new Tuple<int, bool>(3, false);

            Tuple<int, int, int, string> tupla4 = 
                new Tuple<int, int, int, string>(1,2, 3, "Teste");

            Func<int, int, int> Soma = (a, b) => a + b;

            Tuple<Func<int, int, int>, bool> tupla5 =
                new Tuple<Func<int, int, int>, bool>(Soma, false);

            Tuple<bool, int> tupla6 = new Tuple<bool, int>(false, 2);
        }
    }
}
